src/resolvers/__tests__/address.resolvers.spec.ts

```diff
@@ -0,0 +1,136 @@
+import { addressResolver } from '../address.resolvers';
+import * as data from '../__mocks__/resolvers.mocks';
+import dbConnection from '../../config/connectDb';
+import { userResolver } from '../user.resolver';
+
+describe('Address Resolver Tests', () => {
+  let args: any;
+  let superAdminToken: any;
+  let adminToken: any;
+  beforeAll(async () => {
+    await dbConnection;
+    superAdminToken = await userResolver.Mutation.loginUser(null, data.args1);
+    adminToken = await userResolver.Mutation.loginUser(null, data.args2);
+  });
+  afterAll(async () => {
+    dbConnection.close();
+    jest.resetAllMocks();
+  });
+
+  it('Should  create new address', async () => {
+    const result = await addressResolver.Mutation.createAddress(
+      null, data.newAddress, superAdminToken,
+    );
+    expect(result.province).toEqual(data.newAddress.province);
+  });
+
+  it('Should  return error if user not superadmin during address creation', async () => {
+    args = data.newAddress;
+    try {
+      await addressResolver.Mutation.createAddress(null, args, adminToken);
+    } catch (error) {
+      expect(error).toEqual(new Error('Sorry you\'re not authorized to perform this action'));
+    }
+  });
+
+  it('Should  return error if address already exists', async () => {
+    args = data.newAddress;
+    try {
+      await addressResolver.Mutation.createAddress(null, args, superAdminToken);
+    } catch (error) {
+      expect(error).toEqual(new Error('Address already exists! Please check your input and try again'));
```

Comment:
When handling errors what do you think by user apollo error handlings like **AuthenticationError, ForbiddenError, UserInputError, and a generic ApolloError.**?